herbert_core/classes/data_loaders/@spe/private/put_spe.m

```diff
@@ -26,18 +26,16 @@
 filename=[name,ext];
 filepath=[path,filesep];
 
-% Prepare data for Fortran routine: must get rid of NaNs
+% Remove NaNs from data
 index=~isfinite(data.S)|data.S<=null_data|~isfinite(data.ERR);
 if sum(index(:)>0)
     data.S(index)=null_data;
     data.ERR(index)=0;
 end
 
 % Prepare data for Matlab: - must ensure no data has exponents outside range -99 to +99
-% The problem is that although the fortran will write e.g. -1.234E-008 as -1.234-008, which
-% the fortran read routines will cope with, the Matlab and C++ reads will not cope. Matlab
-% sees -1.234-008 as two numbers, and C++ sees this as one number, but does not interpreet the
-% exponent.
+% The problem is that Matlab sees -1.234-008 as two numbers, and C++ sees this 
```

Comment:
this is a problem triggered by a FORTRAN write; which is no longer happening; again do we need this check and clean?

Reply:
My thinking was that I didn't want to change the way we're writing files. I don't know if there are any other routines/workflows that count on the fact that these SPE files do not contain NaNs. I felt leaving these blocks was safer. We also retain backwards compatibility this way.

I think also, `.spe` is being phased out in favour of `.nxspe` anyway, so I don't know if it's worth spending the time working out if it's ok!

Reply:
That's sensible - the only downside is that if we leave the code it will need to be tested.

Reply:
.spe has been phased off, but people still have their old spe around. The code still need to undertstand, so reliability instead of performance is critical. Writing spe still may be requested too for compartibility with old programs and codes. It is so nasty problem -- you do not know what exactly old program users still want so its scary to change anything in this code.

Reply:
If we have an archive of old Horace released versions, we can just tell users who want to use an old SPE file that they can just use the old version of Horace... (otherwise deprecation doesn't mean anything!)

Reply:
That's a good point; it's a massive overhead to try to keep backward compatibility for increasingly rarely used features.

We can make breaking changes so long as we update the major version number and notify our users that it's going to happen.